### ISSUE TITLE
Review usages of `BeforeAll` in tests

### ISSUE THREAD

**Issue by nrmancuso** [checkstyle/checkstyle#12100]
We have discovered that `BeforeAll` used in conjunction with production static field modifications has allowed some surviving mutations that are false positives. From https://github.com/hcoles/pitest/issues/1081#issuecomment-1225447257:

> From a quick scan through the code I think this is "working as expected", as the test suite has hit a wrinkle with how static state works with coverage based mutation systems.

> Because this code is only run if there is no entry in the static BUNDLE_CACHE, the result of mutation testing will be highly dependent on the order in which tests are run. Adding or removing a test may completely change the order.

> If 20 tests could execute the line of code, only the first one to run (and the first ones to run after each time the cache is cleared) will actually execute the code. Only these tests will be linked by coverage to the mutation, so pitest will only select from this reduced set when the mutant is active.

In #12053 , we worked around this issue by using `BeforeEach`:

>@ZuevKirill95 Strangely using `@BeforeEach` instead of `@BeforeAll` kills the mutation, the issue for the same has been opened at https://github.com/hcoles/pitest/issues/1081.

>For now, you can use `@BeforeEach` to kill the mutation.

We should review all usages of `BeforeAll` to see if we have surviving mutations from this issue, and see if we can use `BeforeEach` instead. We should also consider forbidding usage of `BeforeAll` based on our findings, and probably at least restrict future usage of it.

```bash
➜  checkstyle git:(master) ag -Q "@BeforeAll"      
src/test/java/com/puppycrawl/tools/checkstyle/internal/CommitValidationTest.java
97:    @BeforeAll

src/test/java/com/puppycrawl/tools/checkstyle/internal/XpathRegressionTest.java
148:    @BeforeAll

src/test/java/com/puppycrawl/tools/checkstyle/internal/ArchUnitTest.java
35:    @BeforeAll

src/test/java/com/puppycrawl/tools/checkstyle/meta/XmlMetaReaderTest.java
37:    @BeforeAll

src/test/java/com/puppycrawl/tools/checkstyle/AbstractGuiTestSupport.java
38:    @BeforeAll

src/test/java/com/puppycrawl/tools/checkstyle/XpathFileGeneratorAuditListenerTest.java
67:    @BeforeAll

src/it/java/com/google/checkstyle/test/chapter5naming/rule528typevariablenames/MethodTypeParameterNameTest.java
41:    @BeforeAll


```

**Comment by rnveach** [comment]
I think this will connect into an issue with test fields and weither they are allowed to be mutable or not.

https://github.com/checkstyle/checkstyle/pull/12053#discussion_r953876062
> ViolationTest has a static final Locale which gets its value before this method is run. So now we can have a test global variable with its own Locale and then this Before method comes in and set it to its own specific value and possibly conflict with each other. If they both have the same value, there is no issue, but if they are different then this could become messy.

> In my own project, I had to create a rule that "static final fields" for tests must be a single state throughout the entire life of the tests and can never change. I really found this to be helpful when I introduced random test ordering (https://github.com/checkstyle/checkstyle/issues/4818) as I found bad tests that were modifying values that should have been constant throughout the entire test run and was causing unpredictable issues.

**Comment by nick-mancuso** [comment]
> I think this will connect into an issue with test fields and whether they are allowed to be mutable or not.

IMO, I can't see any good coming from mutable test fields.

**Comment by nrmancuso** [comment]
Closed via #12098
